### PR TITLE
Optimize pipeline logic for limit and offset. Remove redundant bitmap code. Fix vector shrink bug.

### PR DESCRIPTION
**Pull request by m-schen**
## What type of PR is this?

- [ ] API-change
- [x] BUG
- [x] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

Cloud-143 and so on.

## What this PR does / why we need it:
1. remove redundant bitmap code which will cause sort slow.
2. fix vector shrink bug which which shrink data to nil and lost mpool information-bytes.
3. fix a pipeline build error which will result in 2 broken-instruction (like order and group) on one pipeline.
4. optimize pipeline build logic which will convert `limit m, offset n` to `top m+n and offset n` now when sum of m+n is less than 8192*4.

### CODE REVIEW

File: pkg/container/vector/vector.go [hunk 1]
```diff
@@ -997,7 +997,7 @@ func ShrinkFixed[T types.FixedSizeT](v *Vector, sels []int64) {
 		vs[i] = vs[sel]
 	}
 	v.Col = vs[:len(sels)]
-	v.data = v.encodeColToByteSlice()
+	v.data = v.data[:len(sels)*v.GetType().TypeSize()]
```
Comment (fengttt) [inline pkg/container/vector/vector.go:1000]: Why is this a fix?   Aren't they doing the same thing?

Comment (m-schen) [reply pkg/container/vector/vector.go:1000]: if `len(sels)` is 0 here. we still need the 16 bytes in front of the source pointer to free the memory.
but encodeColToByteSlice() will return nil. it lost the mpool information. So we get mpool free error finally.

Comment (m-schen) [reply pkg/container/vector/vector.go:1000]: but it's confused for me. if I do this fix, the `free more than alloc` error disappeared.

maybe when we do the encodeColToByteSlice(), go freed the 16 head bytes ?

Comment (m-schen) [reply pkg/container/vector/vector.go]: it's really same if not shrink to 0. I removed the code. 
and add a temporary treatment for shrink to 0.

**Comment by m-schen** [comment]
@matrix-meow rerun failed

**Review by nnsgmsone — APPROVED** [review]

**Pull request merged**